### DISCUSSION TITLE
AC-OPF and NLP

### DISCUSSION THREAD

**Discussion by Iakl** [PyPSA/PyPSA#610] [Q&A]
Hi guys, im wondering if it is possible to solve an NLP AC-OPF problem with PyPSA? I see that it is based on linopy and uses a DC approximation to power flow equations, so I thought it wouldn't. But since PyPSA was based on Pyomo before, maybe it is an option to use the exact power flow equations? Does someone knows about this?

**Comment by fneum** [comment]
Hi @Iakl, we have not implemented AC-OPF (neither in linopy nor pyomo), and I don't think there are any concrete plans to do it. I can recommend pandapower, if you need AC-OPF!
